src/Orchard.Environment.Shell/RunningShellTable.cs

```diff
@@ -33,8 +33,11 @@ public void Add(ShellSettings settings)
                 }
 
                 var hostAndPrefix = GetHostAndPrefix(settings);
+                if (_shellsByHostAndPrefix.ContainsKey(hostAndPrefix))
+                    return false;
                 _shellsByHostAndPrefix.Add(hostAndPrefix, settings);
```

Comment:
maybe it should be just 

_shellsByHostAndPrefix[hostAndPrefix] = settings; 

?